src/core/toga/widgets/table.py

```diff
@@ -3,16 +3,15 @@
 
 
 class TableRow:
-    '''
-    Row of the Table widget
-    '''
     def __init__(self, source, data, icon=None):
-        '''
-        Instantiate a new instance of a row
+        """ Row of the Table widget
+
+        Args:
+            source: The ``class:ListDataSource`` that the row belongs to.
+            data: A ``list`` where each element is a column of the row.
+            icon: A icon displayed in the row.
+        """
 
-        :param data: Information about the row
-        :type  data: ``dict``
-        '''
         self._impl = None
         self.source = source
```

Comment:
This is something that can be cleaned up - you shouldn't need to specify `data` *and* `source` - you should just need data.